core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/V1WorkspaceMigrationMonitor.scala

```diff
@@ -0,0 +1,38 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{ReadAction, WriteAction}
+import org.broadinstitute.dsde.rawls.model.Workspace
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object V1WorkspaceMigrationMonitor {
+
+  import slick.jdbc.H2Profile.api._
+
+  private val v1WorkspaceMigrationHistory: String = "V1_WORKSPACE_MIGRATION_HISTORY"
+
+  final def isMigrating(workspace: Workspace): ReadAction[Boolean] =
+    sql"""SELECT COUNT(*) FROM #${v1WorkspaceMigrationHistory} h
+          WHERE h.WORKSPACE_ID = ${workspace.workspaceId}
+            AND h.STARTED IS NOT NULL
+            AND h.FINISHED IS NULL""".as[Int]
+      .headOption
+      .map(_.getOrElse(0) > 0)
+
+  final def schedule(workspace: Workspace): WriteAction[Unit] =
+    sqlu"""INSERT INTO #${v1WorkspaceMigrationHistory} (WORKSPACE_ID)
```

Comment:
I would probably say the row should be made when we hit an endpoint, not by a monitor. This should probably be updating 'STARTED'?

Reply:
I like it as a message passed to the actor. Here it's not implemented particulary well but I'd say it was the job of the endpoint to enqueue the workspace in the actor's message queue.

Reply:
I think `STARTED` is when migration actually starts. Creating the record schedules the migration attempt (one per attempt). It's up to the actor when it actually starts migration.

Reply:
I like the idea that the database table is private to this monitor. That is enough of an argument for me to do it this way!